File: 공부내용정리/알고리즘/programmers/DFS-BFS/네트워크/NetWork.java
class Solution {
    public int solution(int n, int[][] computers) {
        boolean[] visited = new boolean[n];
        int answer = 0;

        for ( int i = 0; i < n; i++ ) {
            if(!visited[i]) {
                dfs(i, computers, visited, n);
                answer++;
            }
        }

        return answer;

    }

    private void dfs(int node, int[][] computers, boolean[] visited, int n ) {
        visited[node] = true;

        for( int next = 0; next < n; next++ ) {
            if( computers[node][next] == 1 && !visited[next]) {
                dfs(next, computers, visited, n);
            }
        }

    }
}

File: 공부내용정리/알고리즘/programmers/DFS-BFS/네트워크/NetWork.js
function solution(n, computers) {
    const visited = Array(n).fill(false);
    let answer = 0;

    function dfs(node) {
        // 현재 노드 방문 처리
        visited[node] = true;

        // 이 노드와 연결된 다른 노드를 탐색
        for ( let next = 0; next < n; next++ ) {
            // 자기 자신이 아니고, 연결되어있으며 아직 방문 전이면
            if (computers[node][next] === 1 && !visited[next]) {
                dfs(next);
            }
        }
    }

    // 모든 컴퓨터에 대해 방문 체크 여부
    for ( let i = 0; i < n; i++) {
        if(!visited[i]) {
            // i번 컴퓨터를 시작점으로 한 dfs는 i가 속한 네트워크 전체를 방문
            dfs(i);
            answer++; // 하나의 네트워크를 찾았으므로 + 1
        }
    }

    return answer;
}